src/Execution/Execution.php

```diff
@@ -141,6 +156,6 @@ protected function createContext(
      */
     protected function createExecutor(ExecutionContext $context): Executor
     {
-        return new Executor($context, new FieldCollector($context));
+        return new Executor($context, new FieldCollector($context), $this->errorHandler);
```

Comment:
Should we consider refactoring this so we don't have to pass the error handler through this `Execution` thing just to get it into the `Executor`?

Reply:
That might be a bit troublesome because we would need a factory for `Executor` in order to provide both injected and provided arguments to its constructor.